docker/Dockerfile

```
@@ -6,26 +6,19 @@
 # BUILD: docker build --network=host --file docker/Dockerfile --tag pynvml .
 # RUN: docker run --gpus all --rm -it  pynvml
 ##############################################################
-FROM python:3.6
+FROM python:3.8
 
 ENV DEBIAN_FRONTEND noninteractive 
 
-ADD ./requirements.txt /app/requirements.txt
-
-RUN pip install --no-cache-dir --upgrade pip && \
-    pip install --no-cache-dir -r /app/requirements.txt && \
+ADD . /app/.
+RUN cd /app && \
+    pip install --no-cache-dir --upgrade pip && \
+    pip install --no-cache-dir . && \
     rm -rf ~/.cache/pip
```

Think we can write this more cleanly using [Docker's heredoc implementation]( https://www.docker.com/blog/introduction-to-heredocs-in-dockerfiles/ )

Also given the intended usage (`/app` being where `pynvml` is picked up from), maybe we can do a development install using `-e`. Thus adding `/app` to `python`'s search path

Lastly the entrypoint has the following clean up step, which we can just port over here. Plus it needs an update to handle `__pycache__` directories

https://github.com/gpuopenanalytics/pynvml/blob/82cb5e08f56da105a63ca9fcdeb04335bc90950e/docker/entrypoint.sh#L3

All of this together would look like this

```suggestion
RUN <<EOF
cd /app
pip install --no-cache-dir --upgrade pip
pip install --no-cache-dir -e .

find /app -type f -name '*.pyc' -delete
find /app -type d -name '__pycache__' -delete
rm -rf ~/.cache/pip
EOF
```